app-api-common/recipe/src/main/java/org/codeforvictory/android/recipes/data/api/recipe/AdaptToMeal.kt

```diff
@@ -0,0 +1,17 @@
+package org.codeforvictory.android.recipes.data.api.recipe
+
+import org.codeforvictory.android.recipes.data.api.recipe.model.ApiMeal
+import org.codeforvictory.recipes.domain.common.model.meal.Meal
+
+class AdaptToMeal {
```

Comment:
I'd personally prefer the verb Convert.  Also, a pattern I've used in the past that I've liked is having converters specified as extension functions on the Network type.  That lets you name it something like 

```
fun ApiModel.toDomain() : Model 
```

thoughts?

Reply:
The only potential issue I see when using extension functions with anemic modules is the testing side of things. If for any reason we need to mock `ApiModel`, we might encounter a few gotchas when calling `toDomain()`. Does it make sense?

Reply:
I also prefer to have a top-level extension function (`ApiMeal.toMeal()`) to do this conversion. IMO not being able to mock the API models or the conversion logic itself is a feature in the sense that it helps ensure that non-trivial logic doesn't leak into these components.

Reply:
How would you scale adapters/converters if the model had more complex information? The first example that comes to mind is time conversion. If we depend on some other domain model to adjust to commonly expected data formatting, we would have something like:
```
class AdaptToMeal(private clock: Clock) {
	operator fun invoke(apiMeal: ApiMeal): Meal {
        return with(apiMeal) {
            Meal(
                id = id,
                name = name,
                imageThumbUrl = imageThumbnailUrl,
				currentTime = clock.currentTime()
            )
        }
    }
}
```
How would we tackle this problem by using top-level extension functions?

Reply:
My assumption is that I would rarely need to do something complex in this layer since I expect most of the conversions to be presentation-related. So, I would only introduce an adapter when the more complex use case arises and not up front. This argument entirely depends on how the network responses look like in practice, so I'm fine defaulting to this pattern in this repo.